### PR TITLE
Initialize guidance for dev containers

### PR DESCRIPTION
**Pull request by TechnicallyWilliams**
Playbook doesn't have much on dev containers. So for the playbook contribution, I'm contributing two parts which will be represented as two total pull requests. This one introduces dev containers and then the other will be for python specific dev container help. Below is a directory structure code snippet of the expected final outcome:

```bash 
$ tree code-with-engineering-playbook
└───developer-experience
|        ├───devcontainers
|            ├──README.md # first PR
|            └───python
|                └───README.md # second PR
└───observability
|        
```

### CODE REVIEW

File: developer-experience/devcontainers/readme.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+## How can dev containers improve project collaboration?
+
+VS Code dev containers have improved project collaboration between developers on recent team projects by addressing two very specific problems:
+
+- Inconsistent local developer experiences within a team.
+
+- Slow onboarding of developers joining a project.
+
+The problems listed above were addressed by configuring and then sharing a dev container definition. Dev containers are defined by their base image and the artifacts that support that base image. The base image and the artifacts that come with it live in the .devcontainer directory. This directory is where configuration begins. A central artifact to the dev container definition is a json configuration file called `devcontainer`. This file orchestrates the artifacts needed to support the base image and the dev container lifecycle. Installation of the [Remote Development Extension Pack](http://web.archive.org/web/20200828150357/https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack) is required to enable this orchestration within a project repo.
```
Comment (hannesne) [inline developer-experience/devcontainers/readme.md:32]:
```suggestion
The problems listed above were addressed by configuring and then sharing a dev container definition. Dev containers are defined by their base image and the artifacts that support that base image. The base image and the artifacts that come with it live in the .devcontainer directory. This directory is where configuration begins. A central artifact to the dev container definition is a configuration file called `devcontainer.json`. This file orchestrates the artifacts needed to support the base image and the dev container lifecycle. Installation of the [Remote Development Extension Pack](http://web.archive.org/web/20200828150357/https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack) is required to enable this orchestration within a project repo.
```

**Review by charleszipp — COMMENTED** [review]
@TechnicallyWilliams this is a great start. One issue i have with what's here is the lack of connective tissue to what is called out in the root readme. I would like to brainstorm with you when you have some time to figure out how we can better accomplish this. What role do dev containers play in the journey to having a great developer experience and what are some tips or starting points to get there?

File: developer-experience/devcontainers/readme.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Getting Started with Dev Containers
+
+If you are a developer and experienced Visual Studio Code (VS Code) or Docker, then it's probably time you look at [development containers](http://web.archive.org/web/20200806073925/https://code.visualstudio.com/docs/remote/containers) (dev containers). This readme is intended to assist developers in the decision-making process needed to build dev containers. The guidance provided should be especially helpful if you are experiencing VS Code dev containers for the first time.
```
Comment (TessFerrandez) [inline developer-experience/devcontainers/readme.md:3]: nit: experienced Visual Studio Code => have experience with Visual Studio Code

File: developer-experience/devcontainers/readme.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+Let's first introduce what dev containers are, briefly describe how they improve collaboration between developers and then highlight some high-level steps that you will be facing when building a dev container.
```
Comment (TessFerrandez) [inline developer-experience/devcontainers/readme.md:5]: nit: this sentence could be removed and the message would still be the same - if we want to simplify the document

File: developer-experience/devcontainers/readme.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+If the above diagram is not clear, a basic analogy that might help you intuitively understand dev containers is to think of them as a union between Docker’s interactive mode (`docker exec -it 987654e0ff32`) and the VS Code UI experience that you are used to.
+
+To set yourself up for the dev container experience described above, use your VS Code’s Extension Marketplace to install the [Remote Development Extension Pack](http://web.archive.org/web/20200828150357/https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack).
+
+## How can dev containers improve project collaboration?
+
+VS Code dev containers have improved project collaboration between developers on recent team projects by addressing two very specific problems:
+
+- Inconsistent local developer experiences within a team.
+
```
Comment (TessFerrandez) [inline developer-experience/devcontainers/readme.md:29]: nit: remove the empty line

File: developer-experience/devcontainers/readme.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        |   └───.devcontainers # dev container
+        ├───ubuntu
+        |   └───.devcontainers # dev container
+        └───....
+```
+
+Here are the final high-level steps it takes to build a dev container:
+
+1. Decide which platform you'd like to build a local development tool stack around.
+
```
Comment (TessFerrandez) [inline developer-experience/devcontainers/readme.md:78]: nit: remove empty lines in the list

**Review by TessFerrandez — APPROVED** [review]
thanks for the contribution

File: developer-experience/devcontainers/readme.md (hunk 2 above)
Comment (TessFerrandez) [inline developer-experience/devcontainers/readme.md:3]:
```suggestion
If you are a developer and have experience with Visual Studio Code (VS Code) or Docker, then it's probably time you look at [development containers](http://web.archive.org/web/20200806073925/https://code.visualstudio.com/docs/remote/containers) (dev containers). This readme is intended to assist developers in the decision-making process needed to build dev containers. The guidance provided should be especially helpful if you are experiencing VS Code dev containers for the first time.
```

File: developer-experience/devcontainers/readme.md (hunk 4 above)
Comment (TessFerrandez) [inline developer-experience/devcontainers/readme.md:29]:
```suggestion
```

File: developer-experience/devcontainers/readme.md (hunk 5 above)
Comment (TessFerrandez) [inline developer-experience/devcontainers/readme.md:78]:
```suggestion
```

File: developer-experience/devcontainers/readme.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+2. Browse the VS Code provided dev container gallery of project folders that target your platform and choose the most appropriate one.
+
```
Comment (TessFerrandez) [inline developer-experience/devcontainers/readme.md:80]:
```suggestion
```

File: developer-experience/devcontainers/readme.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+3. Inspect the dev container definitions (.devcontainer directory) of a project for the base image and the artifacts that support that base image.
+
```
Comment (TessFerrandez) [inline developer-experience/devcontainers/readme.md:82]:
```suggestion
```

**Review by TessFerrandez — APPROVED** [review]
changed nits to suggestions

**Comment by TessFerrandez** [comment]
@TechnicallyWilliams It has been a while since there has been any activity on this - can I help somehow in the progress?

**Review by brockneedscoffee — APPROVED** [review]

File: developer-experience/devcontainers/readme.md (hunk 3 above)
Comment (TechnicallyWilliams) [reply developer-experience/devcontainers/readme.md]: Thnks. Intro has been shortened.

**Comment by TechnicallyWilliams** [comment]
@charleszipp sync next week to complete this?

**Pull request merged**